vital/kwiversys/SystemTools.cxx

```diff
@@ -4134,6 +4135,34 @@ bool SystemTools::Split(const std::string& str, std::vector<std::string>& lines)
   return true;
 }
 
+
+//----------------------------------------------------------------------------
+/**
+ * Removes duplicate strings in a vector while preserving original order
+ */
+void SystemTools::UniqueOrdered(std::vector<std::string>& items)
+{
+  // For each item: if it has been seen before, then remove it (inplace)
+  std::unordered_set<std::string> seen;
+  std::vector<std::string>::iterator itr = items.begin();
+  while (itr != items.end())
+  {
+    if (!seen.insert((*itr)).second)
+    {
+      // If the item has been seen before, remove it and move itr to the new
+      // location of the next item
+      itr = items.erase(itr);
+    }
+    else
+    {
+      // otherwise go to the next item
+      itr++;
```

Comment:
nit: `++itr` is more efficient. (Not likely to matter here, but as a general guideline, it's good to be in the habit of preferring pre-increment.)